packages/react-native-reanimated/android/src/main/cpp/reanimated/android/NativeProxy.cpp

```diff
@@ -110,6 +110,8 @@ NativeProxy::~NativeProxy() {
   // has already been destroyed when AnimatedSensorModule's
   // destructor is ran
   nativeReanimatedModule_->cleanupSensors();
+
+  layoutAnimations_->cthis()->invalidate();
```

Comment:
I've been dealing a lot with missing invalidations in https://github.com/software-mansion/react-native-reanimated/pull/6850 and I have some thoughts.

Shouldn't we call it from Java when Native Modules are invalidated, for a more transparent flow?

`(java)ReanimatedModule.invalidate() -> ... -> (java)NativeProxy.invalidate() -> (hybrid)LayoutAnimations->invalidate()`

I don't think that relying on Native Proxy's destructor is the best choice here. On the other hand, this is safer since NativeProxy can be "safely" used until it's destroyed.

What do you think?